### DISCUSSION TITLE
[Mia-Platform Console][Design][Microservices] Container ports fallback

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#132] [Q&A] [answered]
Hi guys,
recently you introduced the Containers Ports section in the Microservices page:
<img width="1456" alt="Screenshot 2022-12-12 alle 16 42 06" src="https://user-images.githubusercontent.com/28152586/207295362-ef4f55fc-cb14-4963-8c46-e9954a271bd9.png">

I noticed that if a Marketplace element (plugin, example etc.) doesn't have the `containerPorts` list specified into the configurations JSON, the Target Port will be empty after the creation of that element:
<img width="1456" alt="Screenshot 2022-12-12 alle 16 42 06" src="https://user-images.githubusercontent.com/28152586/207295737-475ffebd-f9a0-42d2-92d9-bcd1baf01bc7.png">

There is a default value for the _Port_ but not for the _Target Port_.
When happens is not so easy find out the problem, can we add a default value for the Target Port too or this behavior is the expected one?
Alternatively a warning message should be great too.

Thanks 😄

**Comment by fredmaggiowski** [comment]
Hi, what marketplace item did cause you problems? we made sure that every item supported by mia-platform had the proper container port already set however some may have slipped out of our control :)

In regards of the default value that is not so easy to assume; what default would you set? based on what? You may be expecting the value `3000` from the `HTTP_PORT` value since it has been an opinionated configuration we are used to, however this is true only for our marketplace items, not every service in the world. We're open to suggestions :)

**Reply by frengooo** [reply]
Hi @fredmaggiowski, a item supported by the client. I personally created it, is the "NestJS Example", but I think I created it before the introduction of this feature, not after.

I agree with you, it's not easy to have a default value, however we need to alert the user in some way, the _Target Port_ can't be empty. Is there the possibility to validate the target port directly on the console, to avoid the commit if it's empty?

**Reply by davidebianchi** [reply]
Could we avoid the default value, and in creation if the port value is not set it is showed as required field to compile. What do you think?

Maybe if you do not know, you can use some casual number, but at least you know what you have configured.

**Reply by fredmaggiowski** [reply]
The port is not required at all, your service may be exposed on the same port (80) making the target port redundant. I agree that the target port may be asked in the service creation form (prefilled with the value from the marketplace item if set) but I'd not make it required.

I'd add the field with a basic explanation on how it should be set and explaining that it can be modified later on. What do you think?

In the meantime I'll add @albertotessarotto to the loop

**Reply by frengooo** [reply]
I agree with @davidebianchi, we can set the _TargetPort_ required and, if someone needs to use the same value for _Port_ and _TargetPort_, he must explain it.

>the target port may be asked in the service creation form

@fredmaggiowski are you talking of the first page, when the user adds the Name of the service, dockerimage and so on?

**Reply by fredmaggiowski** [reply]
Yes the first page... I do not agree with making it required to be honest but we can discuss on this :)

**Reply by fredmaggiowski** [reply, marked as answer]
Brief update: we've planned an update on the feature to provide the user an input during service creation, the idea is to make the user aware of the existence of the port configuration. We hope that it becomes clearer.
